implementations/rust/ockam/ockam_core/src/lib.rs

```diff
@@ -66,3 +66,15 @@ macro_rules! println {
         cortex_m_semihosting::hprintln!($($arg)*).unwrap();
     }};
 }
+
+/// Module for custom implementation of standard traits.
+pub mod traits {
+    use crate::error::Result;
+
+    /// Clone trait for async structs.
+    #[async_trait]
+    pub trait AsyncTryClone: Sized {
+        /// Try cloning a object and return an `Err` if anything panics.
```

Comment:
I think panic is used for different purpose in rust. How about rephrasing it to something like:
```
Try cloning an object and return an `Err` in case of failure.
```